types/index.d.ts

```diff
@@ -62,30 +62,34 @@ export interface AppOptions
   providers: AppProvider[]
 }
 
-export interface CallbacksOptions {
-  signIn?:
-    | (() => true)
-    | ((
-        user: User,
-        account: Record<string, unknown>,
-        profile: Record<string, unknown>
-      ) => Promise<never | string | boolean>)
-  redirect?: (url: string, baseUrl: string) => Promise<string>
-  session?:
-    | ((session: Session) => WithAdditionalParams<Session>)
-    | ((
-        session: Session,
-        userOrToken: User | JWT
-      ) => Promise<WithAdditionalParams<Session>>)
-  jwt?:
-    | ((token: JWT) => WithAdditionalParams<JWT>)
-    | ((
-        token: JWT,
-        user: User,
-        account: Record<string, unknown>,
-        profile: Record<string, unknown>,
-        isNewUser: boolean
-      ) => Promise<WithAdditionalParams<JWT>>)
+export interface Account extends Record<string, unknown> {
+  accessToken: string
+  idToken?: string
+  refreshToken?: string
+  access_token: string
+  expires_in?: number | null
+  refresh_token?: string
+  id_token?: string
+  id: string
+  provider: string
+  type: string
+}
+export interface Profile extends Record<string, unknown> {}
+
+export interface CallbacksOptions<
+  P extends Record<string, unknown> = Profile,
+  A extends Record<string, unknown> = Account
+> {
+  signIn?(user: User, account: A, profile: P): Awaitable<string | boolean>
+  redirect?(url: string, baseUrl: string): Awaitable<string>
+  session?(session: Session, userOrToken: JWT | User): Awaitable<Session>
```

Comment:
This type will force the `session` callback to always take 2 parameters `session, userOrToken`. Are we coolio with that? I  &nbsp;👀&nbsp;  the default callback for the session [is just using one](https://github.com/nextauthjs/next-auth/blob/main/src/server/lib/default-callbacks.js#L46-L48), maybe we should change it to:
```ts
session?:(session: Session, userOrToken?: JWT | User): Awaitable<Session>

Reply:
Yes, so the defulat session callback actually just an identity function and does nothing than return the first param. In any other case, when the user defines/overrides this, they will use the second param to extend session in some way. So I think it's good. 🙂